lib/puppet/application/agent.rb

```diff
@@ -71,6 +71,10 @@ def preinit
     options[:waitforcert] = arg.to_i
   end
 
+  option("--job-id ID") do |arg|
+    options[:job_id] = arg
```

Comment:
Generally +1 just want to confirm there's no way this string could somehow get serialized / deserialized unsafely by the master